### PR TITLE
Renaming infections_mean_varname to infections_varname

### PR DESCRIPTION
**Pull request by gvegayon**
Renames `infections_mean_varname` to `infections_varname` in `latent.infections` and `latent.infectionswithfeedback`.

### CODE REVIEW

**Comment by damonbayer** [comment]
I think `latent_infections_varname` would be clearer, to disambiguate from `observed_infections`.

**Review by damonbayer — DISMISSED** [review]
I think `latent_infections_varname` would be clearer, to disambiguate from `observed_infections`.

**Review by AFg6K7h4fhy2 — COMMENTED** [review]
This follows suit with PR #179 and the change seems fitting. Originally, I thought it might be worth changing `infections_varname` to `latent_infections_varname`, but realized that `latent_infections_varname` might not always be fitting for what is stored in variable.

**Comment by damonbayer** [comment]
> > I think `latent_infections_varname` would be clearer, to disambiguate from `observed_infections`.
> 
> I made a comment at nearly the same time. Are we sure `latent_infections_varname` will always be the most fitting name here?
> 
> @damonbayer

Yes. These are changes in the `latent` sub-module.

**Comment by AFg6K7h4fhy2** [comment]
Ah, yes, I should have remembered more of what's in `./pyrenew/model/`, i.e. `sampled_observed_infections`. 

Given you statement @damonbayer, I also think `infections_varname` &rarr; `latent_infections_varname` is clearer.

**Review by dylanhmorris — COMMENTED** [review]
Now would be a good time to remove the default values for the varnames as well, per #198

**Comment by AFg6K7h4fhy2** [comment]
> Now would be a good time to remove the default values for the varnames as well, per #198

I can do this after our 1-1.

File: model/docs/extending_pyrenew.qmd [hunk 1]
```diff
@@ -163,13 +163,13 @@ class InfFeedback(RandomVariable):
         self,
         infection_feedback_strength: RandomVariable,
         infection_feedback_pmf: RandomVariable,
-        infections_mean_varname: str = "latent_infections",
+        infections_varname: str = "latent_infections",
```
Comment (AFg6K7h4fhy2) [inline model/docs/extending_pyrenew.qmd:166]:
```suggestion
        infections_varname: str = "latent_infections",
```

As per [DHM](https://github.com/CDCgov/multisignal-epi-inference/pull/205#pullrequestreview-2131069672)'s comment.

File: model/src/pyrenew/latent/infections.py [hunk 2]
```diff
@@ -49,14 +49,14 @@ class Infections(RandomVariable):
 
     def __init__(
         self,
-        infections_mean_varname: str = "latent_infections",
+        infections_varname: str = "latent_infections",
```
Comment (AFg6K7h4fhy2) [inline model/src/pyrenew/latent/infections.py:52]:
```suggestion
        infections_varname: str,
```

As per [DHM](https://github.com/CDCgov/multisignal-epi-inference/pull/205#pullrequestreview-2131069672)'s comment.

File: model/src/pyrenew/latent/infections.py [hunk 3]
```diff
@@ -49,14 +49,14 @@ class Infections(RandomVariable):
...
     ) -> None:
         """
         Default constructor for Infections class.
 
         Parameters
         ----------
-        infections_mean_varname : str, optional
+        infections_varname : str, optional
             Name to be assigned to the deterministic variable in the model.
             Defaults to "latent_infections".
```
Comment (AFg6K7h4fhy2) [inline model/src/pyrenew/latent/infections.py:61]:
```suggestion
```

As per [DHM](https://github.com/CDCgov/multisignal-epi-inference/pull/205#pullrequestreview-2131069672)'s comment.

File: model/src/pyrenew/latent/infectionswithfeedback.py [hunk 4]
```diff
@@ -81,7 +81,7 @@ def __init__(
             Infection feedback strength.
         infection_feedback_pmf : RandomVariable
             Infection feedback pmf.
-        infections_mean_varname : str, optional
+        infections_varname : str, optional
             Name to be assigned to the deterministic variable in the model.
             Defaults to "latent_infections".
```
Comment (AFg6K7h4fhy2) [inline model/src/pyrenew/latent/infectionswithfeedback.py:86]:
```suggestion
```

As per [DHM](https://github.com/CDCgov/multisignal-epi-inference/pull/205#pullrequestreview-2131069672)'s comment.

File: model/src/pyrenew/latent/infections.py (hunk 3 above)
Comment (AFg6K7h4fhy2) [inline model/src/pyrenew/latent/infections.py:61]:
```suggestion
        infections_varname : str
            Name to be assigned to the deterministic variable in the model.
```

As per [DHM](https://github.com/CDCgov/multisignal-epi-inference/pull/205#pullrequestreview-2131069672)'s comment.

File: model/src/pyrenew/latent/infectionswithfeedback.py (hunk 4 above)
Comment (AFg6K7h4fhy2) [inline model/src/pyrenew/latent/infectionswithfeedback.py:86]:
```suggestion
        infections_varname : str
            Name to be assigned to the deterministic variable in the model.
```

As per [DHM](https://github.com/CDCgov/multisignal-epi-inference/pull/205#pullrequestreview-2131069672)'s comment.

**Comment by gvegayon** [comment]
Just saw the error. Will fix in a couple of hours

File: model/src/pyrenew/latent/infections.py [hunk 5]
```diff
@@ -48,23 +49,22 @@ class Infections(RandomVariable):
...
             Name to be assigned to the deterministic variable in the model.
-            Defaults to "latent_infections".
 
         Returns
         -------
         None
         """
 
-        self.infections_mean_varname = infections_mean_varname
+        self.latent_infections_varname = latent_infections_varname
```
Comment (gvegayon) [inline model/src/pyrenew/latent/infections.py:67]: Just realized that this is unused @damonbayer. Do we need to use `npro.deterministic` here?

**Review by gvegayon — COMMENTED** [review]
Also, there are a few places where the name of the random variable is hard coded, e.g., infectionswithfeedback.py records the value of Rt_adjusted `npro.deterministic` with a fixed name.

File: model/src/pyrenew/latent/infectionswithfeedback.py [hunk 6]
```diff
@@ -70,7 +69,7 @@ def __init__(
         self,
         infection_feedback_strength: RandomVariable,
         infection_feedback_pmf: RandomVariable,
-        infections_mean_varname: str = "latent_infections",
+        latent_infections_varname: str,
```
[the remark this answers is not in the document]
Comment (damonbayer) [reply model/src/pyrenew/latent/infectionswithfeedback.py:72]: I will add it in here.

File: model/src/pyrenew/latent/infections.py (hunk 5 above)
Comment (damonbayer) [reply model/src/pyrenew/latent/infections.py:67]: actually, we are already saving all_latent_infections in the model classes, so no reason to record them here as well.

File: model/src/pyrenew/latent/infectionswithfeedback.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (damonbayer) [reply model/src/pyrenew/latent/infectionswithfeedback.py:72]: actually, we are already saving all_latent_infections in the model classes, so no reason to record them here as well.

**Comment by damonbayer** [comment]
Upon further inspection, I elected to remove this altogether. We already had a `numpyro.deterministic` sample for `all_latent_infections` (which includes both the `seed_infections` and the `post_seed_infections`), so saving these separately was redundant.

File: model/src/pyrenew/latent/infectionswithfeedback.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dylanhmorris) [reply model/src/pyrenew/latent/infectionswithfeedback.py]: In that case, we should strike this at line 202 as well:

https://github.com/CDCgov/multisignal-epi-inference/blob/2cb49a82eb131748daeef0ee72702c3ccfbc1bbf/model/src/pyrenew/latent/infectionswithfeedback.py#L202

In the longer run, we need to think carefully about when if ever `numpyro.deterministic` should be called below the model level (beyond the scope of this PR).

[the remark this answers is not in the document]
Comment (gvegayon) [reply model/src/pyrenew/latent/infectionswithfeedback.py]: That's what initially made me think. But what was the consensus? I think all variables in deterministic record data using `npro.deterministic`; shouldn't that be the case for all data generated from `RandomVariable.sample()`?

[the remark this answers is not in the document]
Comment (gvegayon) [reply model/src/pyrenew/latent/infectionswithfeedback.py]: I think we should revisit https://github.com/cdcent/cfa-multisignal-renewal/pull/89. I'll cross-post it on the corresponding CDCgov issue.

File: model/docs/example_with_datasets.qmd [hunk 7]
```diff
@@ -165,9 +165,9 @@ I0 = InfectionSeedingProcess(
 # Generation interval and Rt
 gen_int = deterministic.DeterministicPMF(gen_int, name="gen_int")
 rtproc = process.RtRandomWalkProcess(
-    Rt0_dist = dist.TruncatedNormal(loc=1.2, scale=0.2, low=0),
-    Rt_transform = transformation.ExpTransform().inv,
-    Rt_rw_dist = dist.Normal(0, 0.025),
+    Rt0_dist=dist.TruncatedNormal(loc=1.2, scale=0.2, low=0),
+    Rt_transform=transformation.ExpTransform().inv,
+    Rt_rw_dist=dist.Normal(0, 0.025),
```
Comment (AFg6K7h4fhy2) [inline model/docs/example_with_datasets.qmd:170]: This looks fine but I want to know if this was done by `pre-commit` hooks or if you are intentionally following "no spaces padding =" as a personal norm.

File: model/src/pyrenew/latent/infections.py [hunk 8]
```diff
@@ -48,24 +48,15 @@ class Infections(RandomVariable):
...
-        Parameters
-        ----------
-        infections_mean_varname : str, optional
-            Name to be assigned to the deterministic variable in the model.
-            Defaults to "latent_infections".
-
         Returns
         -------
         None
         """
```
Comment (AFg6K7h4fhy2) [inline model/src/pyrenew/latent/infections.py:58]: Any further plans for the `Infections` constructor? The removal of naming variable made me think _looks sparse, are there things we are missing?_.

**Review by AFg6K7h4fhy2 — COMMENTED** [review]
The changes to the naming of variables and the removal of the name variable seem useful. I posed some minor, clarification-seeking comments. I can approve if no one else has further thoughts or questions.

File: model/docs/example_with_datasets.qmd (hunk 7 above)
Comment (damonbayer) [reply model/docs/example_with_datasets.qmd:170]: It was not done by `pre-commit` (see https://github.com/CDCgov/multisignal-epi-inference/issues/159), but it was done by my editor (VS Code).

File: model/src/pyrenew/latent/infections.py (hunk 8 above)
Comment (damonbayer) [reply model/src/pyrenew/latent/infections.py]: Good catch. I will remove it.

File: model/docs/example_with_datasets.qmd (hunk 7 above)
Comment (dylanhmorris) [reply model/docs/example_with_datasets.qmd:170]: Per the [development standards](https://github.com/cdcent/cfa-multisignal-renewal/blob/main/ADR/model/development_standards.md#developing-practices), we follow [PEP 8](https://peps.python.org/pep-0008/) so we [should not have space padding](https://peps.python.org/pep-0008/#other-recommendations) when keyword arguments are assigned in a call.

Comment (AFg6K7h4fhy2) [reply model/docs/example_with_datasets.qmd:170]: @dylanhmorris 

I felt pretty sure `black` was doing this by default, so wasn't expecting to see it as an addition in the PR. There is the possibility that `ruff` followed by `black` in the `pre-commit` config file undoes the change made by `black`. I need to check though and find the issue on the `black` home repo. for others encountering situations where their `black` changes have been reversed by another linter or `pre-commit` hook.

Comment (damonbayer) [reply model/docs/example_with_datasets.qmd:170]: ruff and black formatting from pre-commit do not apply to `qmd` documents. You have to open the file in vscode and format the specific code cell. I must have missed this one at some point, or someone else edited it and did not have formatting enabled in their editor.

**Review by AFg6K7h4fhy2 — APPROVED** [review]
Approved on basis of conversation with @damonbayer.

**Pull request merged**